### PR TITLE
feat: support workload identity

### PR DESCRIPTION
**Pull request by cvvz**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
feat: support workload identity

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
feat: support workload identity
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
feat: support workload identity
```

### CODE REVIEW

**Comment by cvvz** [comment]
TODO: we need to set `useWorkloadIdentityExtension: true` to `/etc/kubernetes/azure.json`

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test
does this PR require updating adal lib version?

**Comment by cvvz** [comment]
> does this PR require updating adal lib version?

No, it's already the latest version

File: pkg/provider/azure.go [hunk 1]
```diff
@@ -931,6 +931,18 @@ func ParseConfig(configReader io.Reader) (*Config, error) {
 	// The resource group name may be in different cases from different Azure APIs, hence it is converted to lower here.
 	// See more context at https://github.com/kubernetes/kubernetes/issues/71994.
 	config.ResourceGroup = strings.ToLower(config.ResourceGroup)
+
+	// these environment variables are injected by workload identity webhook
+	if tenantID := os.Getenv("AZURE_TENANT_ID"); tenantID != "" {
```
Comment (andyzhangx) [inline pkg/provider/azure.go:936]: could you add some ut?

File: pkg/provider/config/azure_auth.go [hunk 2]
```diff
@@ -100,6 +104,28 @@ func GetServicePrincipalToken(config *AzureAuthConfig, env *azure.Environment, r
 		resource = env.ServiceManagementEndpoint
 	}
 
+	if config.UseFederatedWorkloadIdentityExtension {
```
Comment (andyzhangx) [inline pkg/provider/config/azure_auth.go:107]: could you add some ut?

**Review by andyzhangx — COMMENTED** [review]
it lgtm in general, thanks.

**Review by andyzhangx — APPROVED** [review]
pls squash all commits, lgtm

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**